File: backend/src/models/Category.ts
import mongoose from 'mongoose';

const CategorySchema = new mongoose.Schema({
    name: { type: String, required: true },
    description: { type: String, default: '' },
    icon: { type: String, default: 'folder' },
    color: { type: String, default: '#3B82F6' },
    threadCount: { type: Number, default: 0 },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
});

export const Category = mongoose.model('Category', CategorySchema);

File: mobile/app/src/main/java/com/example/derdinesokayim/data/model/Category.kt
package com.example.derdinesokayim.data.model

import androidx.compose.ui.graphics.Color

data class Category(
    val id: String,
    val name: String,
    val description: String,
    val icon: String,
    val color: Color,
    val threadCount: Int = 0
)
